File: Server/models/Setting.js
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

var settingSchema = new Schema({
    user_id: {
        type: Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    manualSetting: {
        type: Boolean,
        default: false
    },
    highTempEnable: {
        type: Boolean,
        default: false
    },
    highTemp: {
        type: Number,
        default: 27.0
    },
    lowTempEnable: {
        type: Boolean,
        default: false
    },
    lowTemp: {
        type: Number,
        default: 15.0
    },
    created_at: {
        type: Date,
        default: Date.now
    },
    updated_at: {
        type: Date,
        default: Date.now
    }
});

var Setting = mongoose.model('Setting', settingSchema);
module.exports = Setting;

File: Client/iOS/IoTSmartHome/IoTSmartHome/Model/Setting.swift
//
//  Setting.swift
//  IoTSmartHome
//
//  Created by Bao Nguyen on 12/21/17.
//  Copyright © 2017 Bao Nguyen. All rights reserved.
//

import UIKit

struct Setting {
    var id: String
    var userId: String
    var manualSetting: Bool
    var highTempEnable: Bool
    var highTemp: Float
    var lowTempEnable: Bool
    var lowTemp: Float
    var created_at: Date
    var updated_at: Date
    
    init(with json: JSON) {
        id = json[KeyString.id] as! String
        userId = json[KeyString.userId] as! String
        manualSetting = json[KeyString.manualSetting] as! Bool
        highTempEnable = json[KeyString.highTempEnable] as! Bool
        highTemp = json[KeyString.highTemp] as! Float
        lowTempEnable = json[KeyString.lowTempEnable] as! Bool
        lowTemp = json[KeyString.lowTemp] as! Float
        created_at = (json[KeyString.created_at] as! String).mongoDBToDate()
        updated_at = (json[KeyString.updated_at] as! String).mongoDBToDate()
    }
}
